lib/lure/command/interfaces.go

```diff
@@ -0,0 +1,28 @@
+package command
+
+import (
+	"github.com/coveooss/lure/lib/lure/project"
+	p "github.com/coveooss/lure/lib/lure/provider"
+	"github.com/coveooss/lure/lib/lure/vcs"
+)
+
+type sourceControl interface {
+	Update(string) (string, error)
+	LogCommitsBetween(string, string) ([]string, error)
+	Branch(string) (string, error)
+	Push() (string, error)
+	WorkingPath() string
+	GetActiveBranches() ([]string, error)
+	CloseBranch(string) error
+	LocalPath() string
+	SanitizeBranchName(string) string
+	Commit(string) (string, error)
+}
+
+type provider interface {
```

Comment:
provider is a pretty generic name, but a good alternative doesn't come to mind...